File: webapp/src/model/status.js
const Status = {
    APPROVED: 'APPROVED',
    WAIT_STATEMENT: 'WAIT_STATEMENT',
    WAIT_COMPANY: 'WAIT_COMPANY',
    WAIT_TRAINING: 'WAIT_TRAINING',
    REGISTER: 'REGISTER',
    values: () => [
        Status.APPROVED,
        Status.WAIT_STATEMENT,
        Status.WAIT_COMPANY,
        Status.WAIT_TRAINING,
        Status.REGISTER
    ],
    has: status => Status.values().includes(status),
    translate: status => {
        return {
            APPROVED: 'Aprovado',
            WAIT_STATEMENT: 'Aguardando envio do termo de resposabilidade',
            WAIT_COMPANY: 'Aguardando recomendação da empresa',
            WAIT_TRAINING: 'Aguardando treinamento',
            REGISTER: 'Registrado',
        }[status];
    },
    next: status => {
        return {
            APPROVED: Status.APPROVED,
            WAIT_STATEMENT: Status.WAIT_TRAINING,
            WAIT_COMPANY: Status.WAIT_STATEMENT,
            WAIT_TRAINING: Status.APPROVED,
            REGISTER: Status.WAIT_COMPANY,
        }[status];
    }
};

export default Status;

File: src/main/java/br/ufsc/cultivar/model/Status.java
package br.ufsc.cultivar.model;

public enum Status {
    APPROVED,
    WAIT_STATEMENT,
    WAIT_COMPANY,
    WAIT_TRAINING,
    REGISTER;

    public boolean isValid(Status next){
      switch (this){
          case APPROVED: return this.equals(next);
          case WAIT_TRAINING: return APPROVED.equals(next) || this.equals(next);
          case WAIT_STATEMENT: return WAIT_TRAINING.equals(next) || this.equals(next);
          case WAIT_COMPANY: return WAIT_STATEMENT.equals(next) || this.equals(next);
          case REGISTER: return WAIT_COMPANY.equals(next) || this.equals(next);
          default: return false;
      }
    }


}
